providers/implementations/signature/eddsa_sig.c

```diff
@@ -326,6 +465,76 @@ static const OSSL_PARAM *eddsa_gettable_ctx_params(ossl_unused void *vpeddsactx,
     return known_gettable_ctx_params;
 }
 
+static int eddsa_set_ctx_params(void *vpeddsactx, const OSSL_PARAM params[])
+{
+    PROV_EDDSA_CTX *peddsactx = (PROV_EDDSA_CTX *)vpeddsactx;
+    const OSSL_PARAM *p;
+
+    if (peddsactx == NULL)
+        return 0;
+    if (params == NULL)
+        return 1;
+
+    p = OSSL_PARAM_locate_const(params, OSSL_SIGNATURE_PARAM_INSTANCE);
+    if (p != NULL) {
+        char instance_name[OSSL_MAX_NAME_SIZE] = "";
+        char *pinstance_name = NULL;
+
+        if (!OSSL_PARAM_get_utf8_string(p, &pinstance_name, sizeof(instance_name)))
+            return 0;
+
+        if (OPENSSL_strcasecmp(pinstance_name, SN_Ed25519_INSTANCE) == 0) {
+            peddsactx->instance_nid = NID_Ed25519_INSTANCE;
+            peddsactx->prehash_flag = 0;
+            peddsactx->context_string_flag = 0;
+        } else if (OPENSSL_strcasecmp(pinstance_name, SN_Ed25519ctx_INSTANCE) == 0) {
+            peddsactx->instance_nid = NID_Ed25519ctx_INSTANCE;
+            peddsactx->prehash_flag = 0;
+            peddsactx->context_string_flag = 1;
+        } else if (OPENSSL_strcasecmp(pinstance_name, SN_Ed25519ph_INSTANCE) == 0) {
+            peddsactx->instance_nid = NID_Ed25519ph_INSTANCE;
+            peddsactx->prehash_flag = 1;
+            peddsactx->context_string_flag = 0;
+        } else if (OPENSSL_strcasecmp(pinstance_name, SN_Ed448_INSTANCE) == 0) {
+            peddsactx->instance_nid = NID_Ed448_INSTANCE;
+            peddsactx->prehash_flag = 0;
+            peddsactx->context_string_flag = 0;
+        } else if (OPENSSL_strcasecmp(pinstance_name, SN_Ed448ph_INSTANCE) == 0) {
+            peddsactx->instance_nid = NID_Ed448ph_INSTANCE;
+            peddsactx->prehash_flag = 1;
+            peddsactx->context_string_flag = 0;
+        } else {
+            // we did not recognize the instance
+            return 0;
+        }
+
+    }
+
+    p = OSSL_PARAM_locate_const(params, OSSL_SIGNATURE_PARAM_CONTEXT_STRING);
+    if (p != NULL) {
+        void *vp_context_string = peddsactx->context_string;
+
+        if (!OSSL_PARAM_get_octet_string(p, &vp_context_string, sizeof(peddsactx->context_string), &(peddsactx->context_string_len))) {
+            peddsactx->context_string_len = 0;
+            return 0;
+        }
+    }
+
+    return 1;
+}
+
+static const OSSL_PARAM settable_ctx_params[] = {
+    OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_INSTANCE, NULL, 0),
+    OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_CONTEXT_STRING, NULL, 0),
+    OSSL_PARAM_END
+};
+
+static const OSSL_PARAM *eddsa_settable_ctx_params(ossl_unused void *vpeddsactx,
+                                                   ossl_unused void *provctx)
+{
+    return settable_ctx_params;
+}
+
 const OSSL_DISPATCH ossl_ed25519_signature_functions[] = {
```

Comment:
I don't think we need separate dispatch tables here since the algorithms are very similar.. Using a mode via a settable seems like the right approach,

Reply:
thanks.  so things are good as I've put them (i.e. you are not suggesting I change something here).